### ISSUE TITLE
[Bug]: Search raise error proxy not healthy，but it was normal before

### ISSUE THREAD

**Issue by xqxq-2020** [milvus-io/milvus#16209]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvusdb/milvus-dev:master-20220325-205c92e5
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): v2.0.1
- OS(Ubuntu or CentOS): CentOS7
- CPU/Memory: Intel(R) Xeon(R) Silver 4215R CPU @ 3.20GHz
- GPU: Quadro K6000
- Others:
```

### Current Behavior

After I start up the 3 required images with docker-compose up -d, I find All services are normal.

But the dashboard show：
![image](https://user-images.githubusercontent.com/62696048/160316927-77f143a1-7fe1-455b-a0c7-6bc7ebddcfb2.png)

And start up the demo programe try error:
![image](https://user-images.githubusercontent.com/62696048/160317037-04e621fb-7545-4eff-9983-910b543cfa55.png)

Then I restart the other two services and it shows that the services still failed：
![image](https://user-images.githubusercontent.com/62696048/160317134-543410c4-fd0d-4c5b-b49b-2ee886605560.png)

![64573bddb4770953d4cc701721878dd](https://user-images.githubusercontent.com/62696048/160317230-4b4026a2-0709-4f9a-8bb5-fdba2ad80fbd.png)

### Expected Behavior

When I start the service, it works fine

### Anything else?

[etcd.log](https://github.com/milvus-io/milvus/files/8359323/etcd.log)
[minio.log](https://github.com/milvus-io/milvus/files/8359324/minio.log)

**Comment by yanliang567** [comment]
@xqxq-2020 the link of etcd.log is broken, could you please re-upload the log? Also could you please upload the log for milvus standalone pod?

/assign @xqxq-2020 
/unassign

**Comment by xqxq-2020** [comment]
[etcd.log](https://github.com/milvus-io/milvus/files/8360082/etcd.log)

milvus standalone pod logfile is so big，i cannot upload it

**Comment by xqxq-2020** [comment]
[milvus-standalone.log](https://github.com/milvus-io/milvus/files/8360097/milvus-standalone.log)

Here is only part of the milvus standalone log

**Comment by yanliang567** [comment]
what is persistent volume of etcd, it is suggested to use SSD for high performance. According to the etcd logs, it seems that it is too slow that rootcoord failed in registering.

**Comment by xqxq-2020** [comment]
To verify whether there is a disk problem, I have restarted the two dockers except etcd, but this error is still reported. 
And we are not using SSD disks at present, and have been purchasing. However, one problem is that I successfully passed the recommendation system project before, but now it is not working

**Comment by yanliang567** [comment]
/assign @LoveEachDay 
@LoveEachDay could you please take a look at this issue?

/unassign

**Comment by LoveEachDay** [comment]
According to the milvus log:
```
[2022/03/28 03:19:45.651 +00:00] [WARN] [client.go:93] ["RootCoordClient mess key not exist"] [key=rootcoord]
[2022/03/28 03:19:45.651 +00:00] [ERROR] [client.go:131] ["failed to get client address"] [error="find no available rootcoord, check rootcoord state"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).connect\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:131\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).GetGrpcClient\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:103\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).callOnce\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:193\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:251\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord/client.(*Client).GetComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/client/client.go:117\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:60\ngithub.com/milvus-io/milvus/internal/util/retry.Do\n\t/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:34\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:84\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:99\ngithub.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:154\ngithub.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:95\ngithub.com/milvus-io/milvus/cmd/components.(*QueryCoord).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/query_coord.go:50\ngithub.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runQueryCoord.func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:192"]
```
milvus cannot find the registered `rootcoord` endpoint from etcd. You can login to the `etcd` container, then run the following command to verify whether key `by-dev/meta/session/rootcoord` exists.
```
etcdctl get --prefix by-dev/meta/session
```
If not, you can run `docker-compose down` to stop the standalone cluster and remove the `volumes` directory. After that, you can run `docker-compose up -d` to start the milvus service again.

**Comment by xqxq-2020** [comment]
After I remove the volumes directory，and run docker-compose up -d to start the milvus service again.

It seems run sucuess:

![8910fe3c4d38ec986fe63705f0e90a5](https://user-images.githubusercontent.com/62696048/160350665-653eafce-1351-4093-951f-f2ff95360931.png)

**Comment by xqxq-2020** [comment]
It seems to be successful, but I want to know why there is a problem, delete the file directory and then restart

**Comment by LoveEachDay** [comment]
Because the data that the milvus standalone cluster are persisted in the `volumes` directory. When the docker container crashes, it will restart using the persisted volumes. And the `metadata` milvus cluster uses may be messed up during several docker restarts. So I suggest that you'd better first check the metadata in the etcd container. Removing the `volumes` directory gives the milvus standalone cluster a refresh start.

**Comment by chang48** [comment]
@LoveEachDay Suppose the persisted `volumes` folder contains my indexed data and I'd like to deploy the container in a different place. If it is necessary to delete the `volumes` folder in order to start the service properly, it seems quite inconvenient to me. Because this seems to imply that one needs to reindex every time the milvus cluster is restarted. Did I miss something?

I'm asking because I am facing a similar issue. I was trying to build a similarity search cluster using haystack's milvus backend. I was able to create the milvus document store (equivalent to the Collection), index data, and update the embeddings. However, I lost all the information every time when I created/initialized a new document store object.
